fastlane/Fastfile

```diff
@@ -86,17 +87,17 @@ platform :ios do
   desc "Testflight build for Live Channel"
   lane :testflight_live do
 
-    updated_version_number_from_version_file = File.read("VERSION").gsub!(/\s+/, '')
+    udpated_version_number = udpate_version_in_xcode_proj(pbx_project_file, get_version_from_file)
 
-    # udpated_version_number = increment_version_number(
-    #   xcodeproj: project,
-    #   version_number: updated_version_number_from_version_file
-    # )
+    ci_run_build_number = ENV["CIRCLE_BUILD_NUM"]
 
-    udpated_version_number = updated_version_number_from_version_file
+    branch_commit_push_and_create_pr_if_needed(
```

Comment:
From what I understood, since you run `udpated_version_number` before this, the only case were there won't be marketing version changes (and therefore no new commit or branch) is if it changed the file with the same version as before, i.e. only a release candidate change, right?

If so, would it make sense to fetch the current version and check it before? Just seems a bit more readable and understandable to me, but let me know if I'm missing something.

To illustrate what I'm suggesting, it would be something like:
```
current_version = get_version_number()
updated_version = get_version_from_file()
ci_run_build_number = ENV["CIRCLE_BUILD_NUM"]

if current_version != updated_version
  udpate_version_in_xcode_proj(pbx_project_file, updated_version)
  branch_commit_push_and_create_pr(updated_version, ci_run_build_number)
  
...
```

That way you could remove all `git diff` matching code as well I believe.

Reply:
Realised after suggesting that there might be an issue executing `get_version_from_file` with inherited values, not sure if selecting a specific target can solve that. Either way I guess there might be other easy options of getting it, but you probably know better if it is not that straight forward 😅

Reply:
Yeah tbh, going through the `pbxproj` is a pain.
Throughout the various experimentations, I went after picking the [pbxproj gem](https://github.com/CocoaPods/Xcodeproj) (got some past positive experience with it). The real problem with it is that the parses can easily break from one version to another of Xcode (in fact it wasn't solving our problem and it was not only updating the version but also messing with the `pbxproj` 's SPM references.
With the implemented method, despite being more convoluted, we have full control of the file itself, without relying on external libraries.
I ended up finding it more stable in the longer term in terms of maintainability.